### PR TITLE
IP2-99-form-module

### PR DESCRIPTION
**Pull request by kmiecikmarcin**

### CODE REVIEW

File: RestSharpProject/Features/FormModule.feature [hunk 1]
```diff
@@ -0,0 +1,91 @@
+﻿Feature: Form module - sign up
+	Description
+	#Task in QA Team: https://tracker.intive.com/jira/browse/IP2-99
```
Comment (katkoch) [inline RestSharpProject/Features/FormModule.feature:3]: This is the issue from JS board, not QA board.

File: RestSharpProject/Features/FormModule.feature [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	And Add Headers
+
+#url to test in zephyr
+Scenario: 1 Send request with required data
+	Given I fill required data
+	When I send the request to API with required data
+	Then The server should return empty body
+
+#url to test in zephyr
+Scenario: 2 Send request filling data and checking all fields
```
Comment (katkoch) [inline RestSharpProject/Features/FormModule.feature:17]: "checking all fields" - which fields do you mean? Technological groups or "zgoda obowiazkowa" and regulamin?

File: RestSharpProject/Features/FormModule.feature [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+Scenario: 7 Send request without marked fields: 'JavaScript, Java, QA, Mobile'
+	Given I fill required data without marked fields: 'JavaScript, Java, QA, Mobile'
+	When I send request to API without marked fields: 'JavaScript, Java, QA, Mobile'
+	Then The server should return error message about unmarked fields: 'JavaScript, Java, QA, Mobile'
+
+#url to test in zephyr
+Scenario: 8 Send request marking all fields from 'Technologie'
+	Given I fill required data with marking all from 'Technologie'
+	When I send request to API marked all technology groups fields 
+	Then The server should return error message about
```
Comment (katkoch) [inline RestSharpProject/Features/FormModule.feature:56]: The end of this sentence is unfinished.

File: RestSharpProject/Features/FormModule.feature [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+#url to test in zephyr
+Scenario: 7 Send request without marked fields: 'JavaScript, Java, QA, Mobile'
+	Given I fill required data without marked fields: 'JavaScript, Java, QA, Mobile'
+	When I send request to API without marked fields: 'JavaScript, Java, QA, Mobile'
+	Then The server should return error message about unmarked fields: 'JavaScript, Java, QA, Mobile'
+
+#url to test in zephyr
+Scenario: 8 Send request marking all fields from 'Technologie'
+	Given I fill required data with marking all from 'Technologie'
+	When I send request to API marked all technology groups fields
```
Comment (katkoch) [inline RestSharpProject/Features/FormModule.feature:55]: Could you rephrase this sentence?

File: RestSharpProject/Features/FormModule.feature [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+Scenario: 9 Send request marking only one field from 'Technologie'
+	Given I fill required data with marking one field about technology groups
+	When I send request to API marked only one field
+	Then The server return empty body
+
+#url to test in zephyr
+Scenario: 10 Send request marking two fields from 'Technologie'
+	Given I fill required data with marking two fields from 'Technologie'
+	When I send request to API marked two fields from 'Technologie'
+	Then The server should execution request and return empty body
```
Comment (katkoch) [inline RestSharpProject/Features/FormModule.feature:68]: The server should **execute** request and return the empty body.

File: RestSharpProject/Features/FormModule.feature [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+	Given I fill required data without fields: 'Hasło' and 'Powtórz hasło'
+	When I send request to API without fields: 'Hasło' and 'Powtórz hasło'
+	Then The server should return error message about missing fields: 'Hasło' and 'Powtórz hasło'
+
+#url to test in zephyr
+Scenario: 12 Send request without repeated password field
+	Given I fill required data without field 'Powtórz hasło'
+	When I send request to API without field about repeated password
+	Then The server should return error message about missing field 'Powtórz hasło'
+
```
Comment (katkoch) [inline RestSharpProject/Features/FormModule.feature:81]: Maybe one more scenario about "hasło", which will be different from "powtórz hasło"?

File: RestSharpProject/Features/FormModule.feature [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+	Then The server should return error message about 
+
+#url to test in zephyr
+Scenario: 9 Send request marking only one field from 'Technologie'
+	Given I fill required data with marking one field about technology groups
+	When I send request to API marked only one field
+	Then The server return empty body
+
+#url to test in zephyr
+Scenario: 10 Send request marking two fields from 'Technologie'
```
Comment (katkoch) [inline RestSharpProject/Features/FormModule.feature:65]: Maybe a scenario with 3 marked technologies would be better? (edge case)

**Review by katkoch — COMMENTED** [review]
I checked your scenarios and made some suggestions.

File: RestSharpProject/Features/FormModule.feature (hunk 2 above)
Comment (kmiecikmarcin) [reply RestSharpProject/Features/FormModule.feature:17]: All this fields

File: RestSharpProject/Features/FormModule.feature (hunk 1 above)
Comment (kmiecikmarcin) [reply RestSharpProject/Features/FormModule.feature]: Done

File: RestSharpProject/Features/FormModule.feature (hunk 2 above)
Comment (kmiecikmarcin) [reply RestSharpProject/Features/FormModule.feature]: Done

File: RestSharpProject/Features/FormModule.feature (hunk 3 above)
Comment (kmiecikmarcin) [reply RestSharpProject/Features/FormModule.feature]: Done

File: RestSharpProject/Features/FormModule.feature (hunk 4 above)
Comment (kmiecikmarcin) [reply RestSharpProject/Features/FormModule.feature:55]: Done

File: RestSharpProject/Features/FormModule.feature (hunk 5 above)
Comment (kmiecikmarcin) [reply RestSharpProject/Features/FormModule.feature]: Done

File: RestSharpProject/Features/FormModule.feature (hunk 6 above)
Comment (kmiecikmarcin) [reply RestSharpProject/Features/FormModule.feature:81]: Done

File: RestSharpProject/Features/FormModule.feature (hunk 7 above)
Comment (kmiecikmarcin) [reply RestSharpProject/Features/FormModule.feature]: Done

File: RestSharpProject/Features/FormModule.feature [hunk 8]
```diff
@@ -1,91 +1,97 @@
...
+	#Task in JS Team: https://tracker.intive.com/jira/browse/IP2-99
 	
 Background: 
 	Given Set the Endpoint with method POST
 	And Add Headers
 
 #url to test in zephyr
-Scenario: 1 Send request with required data
+Scenario: 1 Form module Send request with required data
 	Given I fill required data
```
Comment (AdamAndrysek) [inline RestSharpProject/Features/FormModule.feature:12]: Please observe the Gherkin poll results - if the QA group decides to write Gherkin scenarios in 3rd person, unfortunately you should change it everywhere in your Feature file.

File: RestSharpProject/Features/FormModule.feature [hunk 9]
```diff
@@ -1,91 +1,97 @@
...
+Scenario: 10 Form module Send request marking three fields from 'Technologie'
+	Given I fill required data with marking three fields from 'Technologie'
+	When I send request to API marked three fields from 'Technologie'
+	Then The server should execute request and return empty body
 
 #url to test in zephyr
-Scenario: 11 Send request without fields: 'Hasło' and 'Powtórz hasło'
+Scenario: 11 Form module Send request without fields: 'Hasło' and 'Powtórz hasło'
 	Given I fill required data without fields: 'Hasło' and 'Powtórz hasło'
 	When I send request to API without fields: 'Hasło' and 'Powtórz hasło'
```
Comment (AdamAndrysek) [inline RestSharpProject/Features/FormModule.feature:73]: I'm not sure if the "without fields: 'Hasło' and 'Powtórz hasło'" is necessary here - from Given we know that you don't fill these fields.
If you think you can omit this extract of a sentence, you should also make changes to similar steps in scenarios below and above this one.

File: RestSharpProject/Features/FormModule.feature [hunk 10]
```diff
@@ -1,91 +1,97 @@
...
+Scenario: 14 Form module Send request without marked field 'Zgoda obowiązkowa'
 	Given I fill required data without field 'Zgoda obowiązkowa'
 	When I send request to API without field 'Zgoda obowiązkowa'
 	Then The server should return error message about unmarked field 'Zgoda obowiązkowa'
 
 #url to test in zephyr
-Scenario: 14 Send request with empty form
+Scenario: 15 Form module Send request with empty form
 	Given I send request to API without data
 	Then The server should return error message about missing data
```
Comment (AdamAndrysek) [inline RestSharpProject/Features/FormModule.feature:97]: What do you think about the Scenario when user encounters a server error? 
And what do you think about adding response codes to Then?

**Review by AdamAndrysek — CHANGES_REQUESTED** [review]
My review - some cosmetic changes and suggestions

File: RestSharpProject/Features/FormModule.feature [hunk 11]
```diff
@@ -1,91 +1,97 @@
...
 	Then The server should return error message about unmarked fields: 'JavaScript, Java, QA, Mobile'
 
 #url to test in zephyr
-Scenario: 8 Send request marking all fields from 'Technologie'
-	Given I fill required data with marking all from 'Technologie'
+Scenario: 8 Form module Send request marking all fields about technology groups
+	Given I fill required data with marking all about technology groups
 	When I send request to API marked all technology groups fields 
-	Then The server should return error message about 
+	Then The server should return error message about marked to much technology groups
```
[the remark this answers is not in the document]
Comment (kmiecikmarcin) [reply RestSharpProject/Features/FormModule.feature]: Done

File: RestSharpProject/Features/FormModule.feature [hunk 12]
```diff
@@ -1,91 +1,97 @@
...
 #url to test in zephyr
-Scenario: 6 Send request without field 'Numer telefonu'
+Scenario: 6 Form module Send request without field 'Numer telefonu'
 	Given I fill required data but without field 'Numer telefonu'
 	When I send request to API without field 'Numer telefonu'
 	Then The server should return error message about missing field 'Numer telefonu'
 
 #url to test in zephyr
-Scenario: 7 Send request without marked fields: 'JavaScript, Java, QA, Mobile'
+Scenario: 7 Form module Send request without marked fields: 'JavaScript, Java, QA, Mobile'
```
[the remark this answers is not in the document]
Comment (kmiecikmarcin) [reply RestSharpProject/Features/FormModule.feature]: Done

File: RestSharpProject/Features/FormModule.feature [hunk 13]
```diff
@@ -1,91 +1,97 @@
...
 	When I send request to API marked all technology groups fields 
-	Then The server should return error message about 
+	Then The server should return error message about marked to much technology groups
 
 #url to test in zephyr
-Scenario: 9 Send request marking only one field from 'Technologie'
+Scenario: 9 Form module Send request marking only one field from 'Technologie'
 	Given I fill required data with marking one field about technology groups
 	When I send request to API marked only one field
 	Then The server return empty body
```
[the remark this answers is not in the document]
Comment (kmiecikmarcin) [reply RestSharpProject/Features/FormModule.feature]: Done

File: RestSharpProject/Features/FormModule.feature [hunk 14]
```diff
@@ -1,91 +1,97 @@
...
 	Given I fill all data 
 	When I send request to API 
 	Then Successful execution and the server should return empty body 
 
 #url to test in zephyr
-Scenario: 3 Send request without field 'Imię'
+Scenario: 3 Form module Send request without field 'Imię'
 	Given I fill required data but without field 'Imię'
 	When I send request to API without field 'Imię'
 	Then The server should return error message about missing field 'Imię'
```
[the remark this answers is not in the document]
Comment (kmiecikmarcin) [reply RestSharpProject/Features/FormModule.feature]: I did this into like this: "Then The server should return status 200 with empty JSON body". What do you think?

File: RestSharpProject/Features/FormModule.feature (hunk 8 above)
Comment (kmiecikmarcin) [reply RestSharpProject/Features/FormModule.feature]: Done

File: RestSharpProject/Features/FormModule.feature (hunk 9 above)
Comment (kmiecikmarcin) [reply RestSharpProject/Features/FormModule.feature]: I changed this.

File: RestSharpProject/Features/FormModule.feature (hunk 10 above)
Comment (kmiecikmarcin) [reply RestSharpProject/Features/FormModule.feature]: I added status code. I am not sure about scenarios with server error. I could check only situation when status code in respsonse will not equal to 500.

**Review by katkoch — APPROVED** [review]
The lines are a little long, but I guess it would be hard to make them shorter.
Please correct "Then - line" in Scenario 2.

**Review by AdamAndrysek — APPROVED** [review]
